packages/marshal/src/encodeToCapData.js

```diff
@@ -221,33 +218,30 @@ export const makeEncodeToCapData = ({
         if (qclassMatches(encoded, 'slot')) {
           return encoded;
         }
-        assert.fail(
-          X`internal: Remotable encoding must be an object with ${q(
-            QCLASS,
-          )} ${q('slot')}: ${encoded}`,
-        );
+        // `throw` is noop since `Fail` throws. But linter confused
+        throw Fail`internal: Remotable encoding must be an object with ${q(
```

Comment:
Not a blocker, but I'd like to see if we can fix this before merge. If not the at least having an issue and a caveat comment on Fail.

Reply:
I tried this in [a playground](https://www.typescriptlang.org/play?target=99&noFallthroughCasesInSwitch=true&ts=5.0.0-dev.20221103#code/GYVwdgxgLglg9mABFAFgJzgdwM4AoCUAXImAKYBupaiA3gFCKPLpYmmaICiaGaBdAXzqhIsBMlLYoAVTBpSAQwgoFAIwA2pAIwFaDJqgw4CAbn2NziAPRXk2RMHUKA5vdQx7C++HlKVG0ksIBGw4TQA6dThnXAAiD0QfRWU1TQBCWPxBOjooAE8AB1JEABFSKAUYdXsAXkSwAGswLDAzYLApRAATcsrqxDrcKFIAWwKnYeJwJpaAGkRwxYU0V2IFMDyAbQBdfAGAPkRN4bGJ0nnF8OXXbcQvUt6q7DM6G2RC4tweiqfiMp-qkQ2JQ0MJwNB4EhgH0AGJgL6Pap-RHYIFkEG0ZhGNgcbi8BEAzyeDb4EyIIQiCHiYZSWS+FIBABMunoTAcsPhAHJfKEwJzSZZLG8oPZHC43CgEvckn5UoE2W8etgYPIuncwGrgDARd1JDBnEgUFAoAVsIQbM5tSgQKpwsERlYRjAIBhQsAoFYACofADKLpgBQ9HmwIEkVgAzABWAAMjIAbABiYOh+0jUhgKAAWkjkcZjOjcYAHHmgiEwqRItE4gkZQz0plsq9bPkiogCX0zQ9CXsaod0VQ6O1OtCqgBhEJQAZt74d5Hd4j96i9zGGVhkXE8OB8GdPO7EvKk8lg0SQiS0uTJfyacMsywj9TjjpQXDcxS8-lmNlC5v2LpwSR8pOYrOMwUreBesoBOE36ICMCh5KoxTYKQkiINqe6IPIUAgGgYAwGAIEKIg5AKOooaIHAwDvK2i4wUaJpmhaVo2nacAOk6LpwG6HrekUfpoAGQbYCGYYACwAJziTGpYdOWlYxPE4H0lepAZFkQjHlSSAwn0QyjOMCiTIgnr6WcPpQAJBHYAAgjw8GIAAPogUiWc4OxohQVB6Gyq4cOuXCbnwJwGcM4QAFZwPhBACgIiCaWISA0jIEF1qQom3myOlVAABjyCDZZ+TAwTq95uHADhOCB7hEvUylyjBcEIcUUDlX+iCYFaiD0aa5pWJaqAsfajrOq6lE8b6-qBlYyZhjG4ZxuGSbCSmbFphmmZaIyWjiYWWhaKJokAOwyaEERRApNYpSpanZEAA) and took some notes. It appears to be a Design Limitation. 
https://github.com/microsoft/TypeScript/issues/50363#issuecomment-1219811447

We could file a request for support for never templates so there's a record of this case. WDYT?

Either way, I think the jsdoc for `Fail` should document the limitation and explain when to use `throws` or not.

Reply:
> We could file a request for support for never templates so there's a record of this case. WDYT?

I think that's a good idea. Would you like to?

Reply:
https://github.com/microsoft/TypeScript/issues/50363never mentions templates per se. Is it possible to adapt the workaround explained at https://github.com/microsoft/TypeScript/issues/50363#issuecomment-1219811447 (which I did not understand)?

Reply:
I've filed it here https://github.com/microsoft/TypeScript/issues/51426

Please include it in the jsdoc for the `Fail` function.

Reply:
Done